File: src/dataset/c/sum_of_multiples_Exercism_HeitorMP/gemini/GeminiFlash_sum_of_multiples_v2.c
#include "sum_of_multiples.h"

unsigned int sum(const unsigned int *factors, const size_t number_of_factors,
                 const unsigned int limit) {
  unsigned int result = 0;
  bool multiple[limit];

  if (number_of_factors == 1 && factors[0] == 0) {
    return result;
  }

  for (unsigned int i = 0; i < limit; ++i) {
    multiple[i] = false;
  }

  for (size_t i = 0; i < number_of_factors; ++i) {
    if (factors[i] == 0) continue;

    for (unsigned int j = factors[i]; j < limit; j += factors[i]) {
      multiple[j] = true;
    }
  }

  for (unsigned int i = 0; i < limit; ++i) {
    if (multiple[i]) {
      result += i;
    }
  }

  return result;
}

File: src/dataset/typescript/sum-of-multiples_Exercism-typescript-shybyte/gemini/GeminiFlash_sum-of-multiples_v2.ts
export default (factors: number[]) => {
    return {
        to: (n: number) => {
            let sum = 0;
            for (let i = 3; i < n; i++) {
                let isMultiple = false;
                for (let j = 0; j < factors.length; j++) {
                    if (i % factors[j] === 0) {
                        isMultiple = true;
                        break;
                    }
                }
                if (isMultiple) {
                    sum += i;
                }
            }
            return sum;
        }
    };
};